### PR TITLE
Full pagination feature added.

### PR DESCRIPTION
**Pull request by bhargav-kahar-everestek**

### CODE REVIEW

File: packages/base/Paginator/index.js [hunk 1]
```diff
@@ -15,54 +16,99 @@ const paginationButtonSx = {
...
+  const pagiBtns = [];
+
+  useEffect(() => {
+    setCurrPage(parseInt(offset / pageSize, 10) + 1);
+  }, [offset]);
+
+  if (fullPagintation) {
+    const reminder = parseInt(total / pageSize, 10);
+    const noOfPages =
+      parseInt(total % pageSize, 10) === 0 ? reminder : reminder + 1;
```
Comment (cadejscroggins) [inline packages/base/Paginator/index.js:41]:
```suggestion
    const totalPages = Math.ceil(total / pageSize);
```

File: packages/base/Paginator/index.js [hunk 2]
```diff
@@ -15,54 +16,99 @@ const paginationButtonSx = {
...
-      }
-      onClick={onNextClick}
-      sx={paginationButtonSx}
+}) => {
+  const [currPage, setCurrPage] = useState(1);
+  const pagiBtns = [];
+
+  useEffect(() => {
+    setCurrPage(parseInt(offset / pageSize, 10) + 1);
+  }, [offset]);
```
Comment (cadejscroggins) [inline packages/base/Paginator/index.js:36]: We don't need to keep track of this state. We can calculate it from offset and page size based on what's sent from the parent component.

```suggestion
  const currentPage = Math.floor(offset / pageSize) + 1;
```

File: packages/base/Paginator/index.js [hunk 3]
```diff
@@ -15,54 +16,99 @@ const paginationButtonSx = {
...
+          sx={{
+            ...paginationButtonSx,
+            bg: currPage === i ? 'grays.5' : 'grays.4',
+          }}
+        >
+          {i}
+        </Button>
+      );
+    }
+  }
```
Comment (cadejscroggins) [inline packages/base/Paginator/index.js:60]: If there are 100 pages, doesn't this show all of them? Isn't that bad? I'd think we'd want to show something more like this (same as a Google search):

For page 1: **1** 2 3 4 5 6 7 8 9 10
For page 6: 1 2 3 4 5 **6** 7 8 9 10
For page 7: 2 3 4 5 6 **7** 8 9 10 11
For page 100: 95 96 97 98 99 **100** 101 102 103 104

Maybe the number of pages we show should be configurable? Maybe a `totalJumpToPages` prop with a default of 10?

Comment (bhargav-kahar-everestek) [reply packages/base/Paginator/index.js:60]: That just a basic implementation to working template for Mortis UI. 
Many things needs to be improved still...

File: packages/base/Paginator/index.js [hunk 4]
```diff
@@ -15,54 +16,99 @@ const paginationButtonSx = {
...
-      sx={paginationButtonSx}
+}) => {
+  const [currPage, setCurrPage] = useState(1);
+  const pagiBtns = [];
+
+  useEffect(() => {
+    setCurrPage(parseInt(offset / pageSize, 10) + 1);
+  }, [offset]);
+
+  if (fullPagintation) {
```
[the remark this answers is not in the document]
Comment (bhargav-kahar-everestek) [reply packages/base/Paginator/index.js:38]: Will check for possibility...

[the remark this answers is not in the document]
Comment (bhargav-kahar-everestek) [reply packages/base/Paginator/index.js:38]: The prop `fullPagination` is set from the main page & not the `DashboardPaginator` component.
So, it is good to have this prop. 
If this prop isn't set or false, it works as previous one.

[the remark this answers is not in the document]
Comment (cadejscroggins) [reply packages/base/Paginator/index.js:35]: Ok, if you're going to keep it, can you rename it to `enableJumpToPage`? `fullPagination` is a bit ambiguous to me.

**Pull request merged**